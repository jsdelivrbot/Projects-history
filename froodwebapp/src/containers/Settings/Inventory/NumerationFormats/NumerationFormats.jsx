/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { numericSaveRequest } from 'redux-base/actions';
import { Row, Col } from 'react-flexbox-grid';
import { FormInput, PageHeader, Controls } from 'components';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import { form } from 'styles/common.scss';

const mapDispatchToProps = { numericSaveRequest };

const reduxFormConfig = {
  form: 'numericPageForm',
  // validate: NumericsFormValidation
};

export class NumerationFormatsPage extends Component {
  handleSave = (values) => {
    this.props.numericSaveRequest({
      payload: values
    });
  }

  render() {

    const { handleSubmit } = this.props;

    return (
      <div>
        <PageHeader
          bigText="Adamant and Co. Ltd - Numeration Formats"
        />
        <form
          className={ form }
          onSubmit={ handleSubmit(this.handleSave) }
        >
          <Row middle="xs" center="xs">
            <Col>
              <label>Customer Quotation/Estimate</label>
            </Col>
            <Col xs={ 4 }>
              <Field
                name="quotation"
                placeholder="QU0001"
                component={ FormInput }
              />
            </Col>
          </Row>
          <Row middle="xs" center="xs">
            <Col>
              <label>Sales Orders</label>
            </Col>
            <Col xs={ 4 }>
              <Field
                name="sales"
                placeholder="SQ0001"
                component={ FormInput }
              />
            </Col>
          </Row>
          <Row middle="xs" center="xs">
            <Col>
              <label>Customer Invoices</label>
            </Col>
            <Col xs={ 4 }>
              <Field
                name="invoices"
                placeholder="INV0001"
                component={ FormInput }
              />
            </Col>
          </Row>
          <Row middle="xs" center="xs">
            <Col>
              <label>Shipments</label>
            </Col>
            <Col xs={ 4 }>
              <Field
                name="shipments"
                placeholder="SHP0001"
                component={ FormInput }
              />
            </Col>
          </Row>
          <Row middle="xs" center="xs">
            <Col>
              <label>Purchase Orders</label>
            </Col>
            <Col xs={ 4 }>
              <Field
                name="purchase"
                placeholder="PO0001"
                component={ FormInput }
              />
            </Col>
          </Row>
          <Row middle="xs" center="xs">
            <Col>
              <label>Goods Received Note</label>
            </Col>
            <Col xs={ 4 }>
              <Field
                name="note"
                placeholder="GRN0001"
                component={ FormInput }
              />
            </Col>
          </Row>
          <Controls submitButtonVisible />
        </form>
      </div>
    );
  }
}

NumerationFormatsPage.propTypes = {
  numericSaveRequest: PropTypes.func.isRequired,
  // redux-form related props
  handleSubmit: PropTypes.func.isRequired
};

export default connect(mapDispatchToProps)(reduxForm(reduxFormConfig)(NumerationFormatsPage));
