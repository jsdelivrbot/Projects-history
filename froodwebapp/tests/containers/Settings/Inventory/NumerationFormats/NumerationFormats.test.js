import { NumerationFormatsPage } from 'containers/Settings/Inventory/NumerationFormats/NumerationFormats';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('NumerationFormatsPage', () => {
  const props = {
    numericSaveRequest: jest.fn(),
    handleSubmit: jest.fn()
  };

  it('renders the NumerationFormatsPage component', () => {
    const component = shallow(<NumerationFormatsPage { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of NumerationFormatsPage', () => {
    const component = shallow(<NumerationFormatsPage { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
