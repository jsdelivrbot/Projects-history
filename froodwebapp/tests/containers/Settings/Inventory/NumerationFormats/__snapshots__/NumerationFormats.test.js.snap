// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumerationFormatsPage capturing Snapshot of NumerationFormatsPage 1`] = `
<div>
  <PageHeader
    bigText="Adamant and Co. Ltd - Numeration Formats"
  />
  <form
    className="form"
  >
    <Row
      center="xs"
      middle="xs"
    >
      <Col>
        <label>
          Customer Quotation/Estimate
        </label>
      </Col>
      <Col
        xs={4}
      >
        <Field
          component={[Function]}
          name="quotation"
          placeholder="QU0001"
        />
      </Col>
    </Row>
    <Row
      center="xs"
      middle="xs"
    >
      <Col>
        <label>
          Sales Orders
        </label>
      </Col>
      <Col
        xs={4}
      >
        <Field
          component={[Function]}
          name="sales"
          placeholder="SQ0001"
        />
      </Col>
    </Row>
    <Row
      center="xs"
      middle="xs"
    >
      <Col>
        <label>
          Customer Invoices
        </label>
      </Col>
      <Col
        xs={4}
      >
        <Field
          component={[Function]}
          name="invoices"
          placeholder="INV0001"
        />
      </Col>
    </Row>
    <Row
      center="xs"
      middle="xs"
    >
      <Col>
        <label>
          Shipments
        </label>
      </Col>
      <Col
        xs={4}
      >
        <Field
          component={[Function]}
          name="shipments"
          placeholder="SHP0001"
        />
      </Col>
    </Row>
    <Row
      center="xs"
      middle="xs"
    >
      <Col>
        <label>
          Purchase Orders
        </label>
      </Col>
      <Col
        xs={4}
      >
        <Field
          component={[Function]}
          name="purchase"
          placeholder="PO0001"
        />
      </Col>
    </Row>
    <Row
      center="xs"
      middle="xs"
    >
      <Col>
        <label>
          Goods Received Note
        </label>
      </Col>
      <Col
        xs={4}
      >
        <Field
          component={[Function]}
          name="note"
          placeholder="GRN0001"
        />
      </Col>
    </Row>
    <div>
      error
    </div>
    <Controls
      submitButtonVisible={true}
    />
  </form>
</div>
`;
